samples/net/sockets/echo_server/boards/atsamr21_xpro.conf

```diff
@@ -1,23 +1,31 @@
 #
-# Copyright (c) 2020-2021, Gerson Fernando Budke
 # Copyright (c) 2019, Benjamin Valentin
+# Copyright (c) 2020-2023, Gerson Fernando Budke
 #
 # SPDX-License-Identifier: Apache-2.0
 #
 
-# Reduced buffers to fit into SAMR21 SoC
+# Reduced configs to fit into SAMR21 SoC
 CONFIG_CPP=n
 
-CONFIG_NET_PKT_RX_COUNT=6
-CONFIG_NET_PKT_TX_COUNT=6
-CONFIG_NET_BUF_RX_COUNT=6
-CONFIG_NET_BUF_TX_COUNT=6
-CONFIG_NET_MAX_CONTEXTS=4
+CONFIG_NET_PKT_RX_COUNT=5
+CONFIG_NET_PKT_TX_COUNT=5
+CONFIG_NET_BUF_RX_COUNT=5
+CONFIG_NET_BUF_TX_COUNT=5
+CONFIG_NET_MAX_CONTEXTS=2
 CONFIG_NET_MAX_CONN=1
 CONFIG_NET_MAX_ROUTES=1
 CONFIG_NET_MAX_NEXTHOPS=1
 
-CONFIG_SHELL_STACK_SIZE=768
-CONFIG_SHELL_CMD_BUFF_SIZE=64
+CONFIG_SHELL_STACK_SIZE=384
+CONFIG_SHELL_CMD_BUFF_SIZE=16
 CONFIG_SHELL_ARGC_MAX=6
-CONFIG_SHELL_HISTORY_BUFFER=64
+CONFIG_SHELL_HISTORY_BUFFER=4
+
+CONFIG_LOG_MODE_MINIMAL=y
+CONFIG_NET_IP_DSCP_ECN=n
+CONFIG_NET_STATISTICS=n
+CONFIG_NET_MGMT_EVENT_STACK_SIZE=384
```

Comment:
This looks too very low, but might work ok. Note that there is `CONFIG_NET_DEBUG_MGMT_EVENT_STACK` option which prints stack usage after each event, and that can be used to verify the numbers.

Reply:
```
*** Booting Zephyr OS build zephyr-v3.5.0-5222-g03772eb0d806 ***


uart:~$ I: Initializing network
I: IPv6 address: 2001:db8::1
I: Run echo server
I: Network connected
I: Starting...
I: Waiting for UDP packets on port 4242 (IPv6)...
D: (udp6): UDP (IPv6): Received and replied with 21 bytes
D: (udp6): UDP (IPv6): Received and replied with 46 bytes
D: (udp6): UDP (IPv6): Received and replied with 12 bytes
D: (udp6): UDP (IPv6): Received and replied with 54 bytes
D: (udp6): UDP (IPv6): Received and replied with 84 bytes
D: (udp6): UDP (IPv6): Received and replied with 46 bytes
D: (udp6): UDP (IPv6): Received and replied with 65 bytes
D: (udp6): UDP (IPv6): Received and replied with 3 bytes
D: (udp6): UDP (IPv6): Received and replied with 116 bytes
D: (udp6): UDP (IPv6): Received and replied with 24 bytes
D: (udp6): UDP (IPv6): Received and replied with 95 bytes
D: (udp6): UDP (IPv6): Received and replied with 25 bytes
D: (udp6): UDP (IPv6): Received and replied with 67 bytes
D: (udp6): UDP (IPv6): Received and replied with 97 bytes
D: (udp6): UDP (IPv6): Received and replied with 108 bytes
```

```
*** Booting Zephyr OS build zephyr-v3.5.0-5222-g03772eb0d806 ***


uart:~$ I: Initializing network
I: IPv6 address: 2001:db8::2
I: Run echo client
I: Network connected
I: Starting...
D: IPv6 UDP: Sent 21 bytes
D: IPv6 UDP: Received and compared 21 bytes, all ok
D: IPv6 UDP: Sent 46 bytes
D: IPv6 UDP: Received and compared 46 bytes, all ok
D: IPv6 UDP: Sent 12 bytes
D: IPv6 UDP: Received and compared 12 bytes, all ok
D: IPv6 UDP: Sent 54 bytes
D: IPv6 UDP: Received and compared 54 bytes, all ok
D: IPv6 UDP: Sent 84 bytes
D: IPv6 UDP: Received and compared 84 bytes, all ok
D: IPv6 UDP: Sent 46 bytes
D: IPv6 UDP: Received and compared 46 bytes, all ok
D: IPv6 UDP: Sent 65 bytes
D: IPv6 UDP: Received and compared 65 bytes, all ok
D: IPv6 UDP: Sent 3 bytes
D: IPv6 UDP: Received and compared 3 bytes, all ok
D: IPv6 UDP: Sent 116 bytes
D: IPv6 UDP: Received and compared 116 bytes, all ok
D: IPv6 UDP: Sent 24 bytes
D: IPv6 UDP: Received and compared 24 bytes, all ok
D: IPv6 UDP: Sent 95 bytes
D: IPv6 UDP: Received and compared 95 bytes, all ok
D: IPv6 UDP: Sent 25 bytes
D: IPv6 UDP: Received and compared 25 bytes, all ok
D: IPv6 UDP: Sent 67 bytes
D: IPv6 UDP: Received and compared 67 bytes, all ok
D: IPv6 UDP: Sent 97 bytes
D: IPv6 UDP: Received and compared 97 bytes, all ok
D: IPv6 UDP: Sent 108 bytes
D: IPv6 UDP: Received and compared 108 bytes, all ok
```